### DISCUSSION TITLE
TIF Compression

### DISCUSSION THREAD

**Discussion by GenuineRex** [dlemstra/Magick.NET#1116] [Help]
I am opening a TIF file with new MagickImage(""); and then setting the compression to CompressionMethod.JPEG and saving as a TIF.

The resulting TIF file is 3-4x larger than the original which I believe has the same type of compression.

Is there something else I need to set to get better compression?

Thanks in advance!
Rex

**Comment by GenuineRex** [comment]
I was able to get close with  tif.Quality = 11;

I am trying to write a watermark on a TIF file that I did not create and write out the same characteristics of the TIF I read.

Are there any other settings I should consider?

Rex

**Comment by dlemstra** [comment]
Can you share your input image and the watermark?

**Reply by GenuineRex** [reply]
Unfortunately the image has sensitive data.  Hard to even create a sample with similar qualities because these are customer-generated on their scanner.

What are the properties to consider when trying to achieve similar quality and compression to an original file?

tif.PreserveFileSettings() would be a handy function.  I am pulling the TIF in from a ZIP file stream into a memory stream and a FileCollection.  Then I iterate each file to watermark and set compression and Quality.  I don't really want to change the compression and quality but because I am needing to modify and then save the TIF I must.

**Reply by dlemstra** [reply]
The primary setting is `image.Settings.Compression`. And this is how it works for the `tiff` coder:

When you set `image.Settings.Compression` that will be used. Otherwise when the `Compression` of the `image` is not `jpeg` that will be used. The quality setting is only used when `jpeg` or `zip` compression is used, but you will need to explicitly set it in the `image.Settings`.
